src/github.rs

```diff
@@ -112,12 +112,23 @@ impl<'a> ToUrl for CommitsUrl<'a> {
 
 impl<'a> ToUrl for SingleCommitUrl<'a> {
     fn url(&self) -> String {
-        format!(
-            "https://api.github.com/repos/{OWNER}/{REPO}/commits/{REF}",
-            OWNER = OWNER,
-            REPO = REPO,
-            REF = self.sha
-        )
+        // "origin/master" is set as `sha` when there is no `--end=` definition
+        // specified on the command line.  We define the GitHub master branch
+        // HEAD commit as the end commit in this case
+        if self.sha == "origin/master" {
+            format!(
+                "https://api.github.com/repos/{OWNER}/{REPO}/commits/HEAD",
```

Comment:
would `commits/master` here work just as well?

if so, I think that would be clearer. (I'll be honest, I'm not clear on what `HEAD` is supposed to denote when querying `github`...)

Reply:
> would commits/master here work just as well?

Yes it looks like it:

https://api.github.com/repos/rust-lang/rust/commits/master

https://api.github.com/repos/rust-lang/rust/commits/HEAD

Reply:
Pushed this change in be54918d5e80f1cccc18de0e7885a1101d74f1c2